hld/config/config.go

```diff
@@ -0,0 +1,106 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/viper"
+)
+
+// Config represents the daemon configuration
+type Config struct {
+	// Socket configuration
+	SocketPath string `mapstructure:"socket_path"`
+
+	// API configuration (for future phases)
+	APIKey     string `mapstructure:"api_key"`
+	APIBaseURL string `mapstructure:"api_base_url"`
+
+	// Logging configuration
+	LogLevel string `mapstructure:"log_level"`
+}
+
+// Load loads configuration with priority: flags > env vars > config file > defaults
+func Load() (*Config, error) {
+	v := viper.New()
+
+	// Set config name and paths
+	v.SetConfigName("daemon")
+	v.SetConfigType("json")
+
+	// Add config paths in order of preference
+	v.AddConfigPath(".")                                                       // Current directory
+	v.AddConfigPath(getDefaultConfigDir())                                     // XDG config directory
+	v.AddConfigPath(filepath.Join(os.Getenv("HOME"), ".config", "humanlayer")) // Fallback config directory
+
+	// Set environment variable prefix and automatic env reading
+	v.SetEnvPrefix("HUMANLAYER")
+	v.AutomaticEnv()
+
+	// Map environment variables to config keys
+	_ = v.BindEnv("socket_path", "HUMANLAYER_DAEMON_SOCKET")
+	_ = v.BindEnv("api_key", "HUMANLAYER_API_KEY")
```

Comment:
does this not happen automatically?

basically curious why we need BindEnv vs. just having key names that fit viper's expectation of how env maps to config file name etc...